### PR TITLE
Add which command using npm module which

### PR DESCRIPTION
**Pull request by khuongduybui**

### CODE REVIEW

**Comment by khuongduybui** [comment]
I vetted the source code of the `which` npm module myself to make sure it's a cross platform, pure-JS implementation.

**Comment by nfischer** [comment]
Haven't tested this yet, but I'm in favor of re-using existing modules if they're robust enough.

@khuongduybui does this also support things like:

```bash
$ export foo="cat"
$ which $foo # outputs path to cat
```

If so, could you add tests to the pre parser test file?

File: test/which.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const results = cash.which();
+    console.log(typeof results);
+    results.should.equal('');
+  });
+
+  it('should find cmd when called', function () {
+    const results = cash.which('cmd');
+    console.log(typeof results);
+    results.should.not.equal('');
+  });
```
Comment (dthree) [inline test/which.js]: It should be pretty easy tests for basic functionality. Not trying to re-handle all of the tests that the lib already provides, but a basic test or two that verifies the author didn't butcher the package on the latest update would be good.

File: src/help/which.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = `
+Usage: which COMMAND
+Look for COMMAND in $PATH.
+
+      --help                   display this help and exit
+
+Report which bugs to <https://github.com/khuongduybui/cash>
```
Comment (dthree) [inline src/help/which.js]: Should probably leave this upstream at `https://github.com/dthree/cash` for consistency and one source of all bugs for the project.

File: src/commands/which.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  vorpal
+    .command('which [command]')
+    .parse(preparser)
+    // .option('-o, --option', 'option description')
+    .action(function (args, callback) {
+    //   args.options = args.options || {};
+      return interfacer.call(this, {
+        command: which,
+        args: args.command, // only pass in what you need from Vorpal
+        // options: args.options, // split the options into their own arg
```
Comment (dthree) [inline src/commands/which.js]: Could you alter this to `options: {},` if no options needed?

File: src/commands/which.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+module.exports = function (vorpal) {
+  if (vorpal === undefined) {
+    return which;
+  }
+  vorpal.api.which = which;
+
+  vorpal
+    .command('which [command]')
+    .parse(preparser)
+    // .option('-o, --option', 'option description')
```
Comment (dthree) [inline src/commands/which.js]: Just toss the commented lines.

File: commands.json [hunk 5]
```diff
@@ -22,7 +22,8 @@
     "less",
     "grep",
     "rm",
-    "unalias"
+    "unalias",
+    "which"
     ],
   "globalExclusions": [
     "alias",
```
Comment (dthree) [inline commands.json]: Which should be able to be global too, which you can simply do by adding it lower on as well, down here:

https://github.com/dthree/cash/blob/master/commands.json#L171

**Comment by dthree** [comment]
Hey looks great, thanks!!!

Added some notes. After this, should be good to merge!

File: commands.json (hunk 5 above)
Comment (khuongduybui) [inline commands.json]: And gulp will generate packages/which for me?

**Comment by khuongduybui** [comment]
@nfischer: it does work, I'll add a test.
@dthree: thanks for quick response, I will make the suggested changes

**Comment by nfischer** [comment]
What does this return in the case of `which ls`? Does this indicate the `ls` implementation inside cash, or does this return the path to `ls` on the system? Not sure which behavior would be better in our case, but it's something to think about (and might be worth clarifying in the documentation).

I could see a user being confused about why `which ls` returns `/bin/ls` when executing `ls` actually execute's cash's `ls` command. But then again, we see the same issue in Bash, since `which pwd` refers to `/bin/pwd` and yet the `pwd` command [usually calls a builtin](https://stackoverflow.com/questions/3134479/shell-built-in-pwd-versus-bin-pwd).

**Comment by dthree** [comment]
Should probably just do the system one - if Cash is installed on Windows globally, it will return the Cash one anyway.

**Comment by khuongduybui** [comment]
@nfischer: I'm with @dthree here. Wikipedia (which we all agree is the ultimate source of truth, don't we?) says "which is a Unix command used to identify the location of executables." This which command does exactly that.

File: commands.json (hunk 5 above)
Comment (nfischer) [inline commands.json]: @khuongduybui yeah, gulp should generate everything for you. It should be easy--if it's not, let us know and we can work on fixing it.

**Comment by nfischer** [comment]
Yeah, I'd agree it's probably the better route (I personally hate the `which` command for this specific reason). This might motivate implementing a command like `type` to provide nicer semantics (identifies builtins).

**Comment by khuongduybui** [comment]
@dthree: ready for merge

**Comment by dthree** [comment]
Thanks! Will do final review shortly.

**Comment by dthree** [comment]
:+1: sorry for the delay

**Pull request merged**